financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheetViewModel.kt

```diff
@@ -161,11 +185,25 @@ internal class FinancialConnectionsSheetViewModel @Inject constructor(
      * the back button or closed the custom tabs UI, so return result as canceled.
      */
     internal fun onBrowserActivityResult() {
-        setState {
-            if (webAuthFlowActive && activityRecreated) {
-                copy(viewEffect = FinishWithResult(Canceled))
-            } else {
-                this
+        viewModelScope.launch {
+            mutex.withLock {
+                setState {
```

Comment:
@carlosmuvi-stripe should this also not be async since it's inside a mutex lock?